server/container/guice/protocols/jmap/src/main/java/org/apache/james/modules/protocols/JmapEventBusModule.java

```diff
@@ -20,14 +20,18 @@
 package org.apache.james.modules.protocols;
 
 import org.apache.james.events.EventBus;
+import org.apache.james.events.EventListener;
 import org.apache.james.jmap.InjectionKeys;
+import org.apache.james.jmap.pushsubscription.PushListener;
 
 import com.google.inject.AbstractModule;
+import com.google.inject.multibindings.Multibinder;
 import com.google.inject.name.Names;
 
 public class JmapEventBusModule extends AbstractModule {
     @Override
     protected void configure() {
         bind(EventBus.class).annotatedWith(Names.named(InjectionKeys.JMAP)).to(EventBus.class);
+        Multibinder.newSetBinder(binder(), EventListener.ReactiveGroupEventListener.class).addBinding().to(PushListener.class);
```

Comment:
Replace this by

```
    @ProvidesIntoSet
    InitializationOperation registerPushListener(@Named(InjectionKeys.JMAP) EventBus jmapEventBus, PushListener pushListener) {
        return InitilizationOperationBuilder
            .forClass(PushListener.class)
            .init(() -> jmapEventBUs.register(pushListener));
    }
```

Reply:
I registered it in `org.apache.james.modules.event.JMAPEventBusModule`  (distributed package)

```
    @ProvidesIntoSet
    InitializationOperation workQueue(@Named(InjectionKeys.JMAP) RabbitMQEventBus instance, PushListener pushListener) {
        return InitilizationOperationBuilder
            .forClass(RabbitMQEventBus.class)
            .init(() -> {
                instance.start();
                instance.register(pushListener);
            });
    }
```

It also works, but I don't sure it is good code convention